Moved batch to end of processors pipeline

harshit-splunk opened the pull request:
Addresses https://github.com/signalfx/splunk-otel-collector-chart/issues/455

jvoravong commented on helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:
```diff
@@ -158,8 +158,11 @@ exporters:
+      level: {{ .Values.logLevel }}
     metrics:
       address: 0.0.0.0:8889
+
```
Nit: Extra new line.

harshit-splunk commented:
> We have a fluentd.config.logLevel config, so I don't recommend we have a global logLevel config. Maybe in another PR add an example of how to enable debug logging like how our docs describe.

Maybe we can depreciate `fluentd.config.logLevel` and use global logLevel config in both fluentd and otel.

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:
```diff
@@ -646,6 +645,7 @@ service:
         {{- if .Values.environment }}
         - resource/add_environment
         {{- end }}
+        - batch
```
Please move it right after filter processor. There is no performance improvement in moving it further

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:
```diff
@@ -673,11 +673,11 @@ service:
         {{- if .Values.environment }}
         - resource/add_environment
         {{- end }}
+        - batch
```
No change needed here

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:
```diff
@@ -699,12 +699,12 @@ service:
         {{- if .Values.environment }}
         - resource/add_environment
         {{- end }}
+        - batch
```
Same. No need to change it here. Resource enrichment processors should work better on batches

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:
```diff
@@ -732,6 +731,7 @@ service:
         {{- if .Values.isWindows }}
         - metricstransform
         {{- end }}
+        - batch
```
Same here

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:
```diff
@@ -751,10 +751,10 @@ service:
         - resource/add_agent_k8s
         - resourcedetection
         - resource
+        - batch
```
and here

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:
```diff
@@ -186,6 +185,7 @@ service:
         {{- if .Values.environment }}
         - resource/add_environment
         {{- end }}
+        - batch
```
no change needed here as well

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:
```diff
@@ -195,14 +195,14 @@ service:
         {{- if (and .Values.splunkPlatform.metricsEnabled .Values.environment) }}
         - resource/add_environment
         {{- end }}
+        - batch
```
same

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:
```diff
@@ -227,9 +227,9 @@ service:
-        - batch
         - filter/logs
         - resource/logs
+        - batch
```
Please move it right after `filter/logs`

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-collector.tpl:
```diff
@@ -245,10 +245,10 @@ service:
         - resource/add_collector_k8s
         - resourcedetection
         - resource/add_cluster_name
+        - batch
```
No change needed here

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-k8s-cluster-receiver-config.tpl:
```diff
@@ -188,7 +188,7 @@ service:
     metrics:
       receivers: [k8s_cluster]
-      processors: [memory_limiter, batch, resource, resource/k8s_cluster]
+      processors: [memory_limiter, resource, resource/k8s_cluster, batch]
```
Same. No changes needed in this file

dmitryax commented:
Please revert most of the changes unless you can prove performance improvement with some benchmarks

harshit-splunk commented:
I have reverted the changes. I have converted this PR to a draft. I am not sure how I can benchmark the pipeline. Do you have any references for how we have done it in past?

atoulme commented:
You might want to look at https://github.com/signalfx/splunk-otel-collector-chart/tree/main/test to get started, and see if you can add a test that emulates enough load.

dmitryax commented on helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl:
```diff
@@ -634,9 +634,9 @@ service:
-        - batch
         {{- if not $gateway.enabled }}
         - filter/logs
+        - batch
         {{- end }}
```
```suggestion
        {{- end }}
        - batch
```

dmitryax commented:
I mean we need benchmarks if you want to push the batch processor to the very end. I don't really see a reason to do that

dmitryax commented:
Please rebase
